packages/headless/src/features/folding/folding-slice.ts

```diff
@@ -172,23 +172,12 @@ export const foldingReducer = createReducer(
       .addCase(
         loadCollection.fulfilled,
         (state, {payload: {collectionId, results}}) => {
-          const rootResult = resolveRootFromFields(
+          const newCollections = createCollections(
             results as ResultWithFolding[],
             state.fields
           );
-          if (!rootResult) {
-            return;
-          }
```

Comment:
With the new approach, is it guaranteed that the root will always exist? Or is there [an assumption](https://github.com/coveo/ui-kit/pull/1199/files#diff-e8a038157bf1f9e52c95676be3d14f2ca6a80934f9cfe2b52c6d0f5bb06d8ea5R90) that the root is in the top 100?

Reply:
It will always be there, unless the index completely breaks the parent-child loading feature as a whole.